### PR TITLE
Final changes.

### PR DESCRIPTION
**Pull request by Alejandroq12**
Hello, dear reviewr.

# Pull Request: Ruby on Rails Capstone Project - Budget App

## Summary of Changes
- Implemented user registration and login using the Devise gem for authentication.
- Created pages for home, sign up, log in, transactions, and add new category/transaction.
- Ensured the application follows Rails RESTful design and router.
- Utilized controllers to handle requests and render views.
- Integrated params from browser requests in a secure manner.
- Applied layouts and templates for shared content.
- Used database migration files to maintain the database schema.
- Implemented validations for models to enhance data integrity.
- Addressed n+1 problems to optimize database queries.
- Utilized ActiveRecord for writing SQL queries.
- Set up associations between models according to the ERD diagram.
- Limited access to webapp resources based on authorization rules.
- Conducted unit and integration tests for critical components.
- Documented the project in a professional way, adhering to gitflow principles.

## Unimplemented Features
- **Hamburger button**: Not implemented due to time constraints as today is the project deadline.

## Additional Notes
- The video presentation was recorded quicly due to the deadline and may lack in detail, please take it into account.
- All functionalities have been implemented according to the project's design guidelines except for the hamburger menu.

## Learning Outcomes
- Gained proficiency in using Ruby gems and the Ruby on Rails framework.
- Enhanced understanding of Rails RESTful design, ORM, and ActiveRecord.
- Improved skills in using layouts, templates, and database migrations.
- Reinforced the importance of testing and documentation in software development.

## Challenges
- Time management was a significant challenge, leading to the omission of the hamburger menu.
- Strived to maintain a balance between feature completion and code quality under time pressure.

Please review the changes and provide feedback.

Please take into account that I added the functionality to select different categories only to the button that it is in the main page:
In here a movement may be created for different categories. And, the another way to create a movement is to click a group and in that case the movement will belong to that specific group/category.
https://pocket-guardian-fec21289dfe6.herokuapp.com/movements/new
image

Screenshot from 2024-03-02 20-30-16
And the button inside each group it assigns the transaction to that specific group:
Screenshot from 2024-03-02 21-08-52
Screenshot from 2024-03-02 21-11-44

The two buttons were created to add a movement but they conditionally act different to meet the requirement in a good way.
Please, note that for signing up you need a real email in order to receive the confirmation email.

### CODE REVIEW

File: app/views/groups/new.html.erb
Comment (Efiamotu-1) [inline app/views/groups/new.html.erb:1]: - [ ] From the project requirement you are to create a `category page` as well as a `Add new category` button, but all of this is missing from your project, please kindly fix this as soon as possible so you can be approved.
anc r
anc
- [ ] The add new category button should be placed at the bottom of the page.

Comment (Efiamotu-1) [inline app/views/groups/new.html.erb:1]: - [ ] From the project requirement you are to create a `transactions page` as well as a `Add new transaction` button, but all of this is missing from your project, please kindly fix this as soon as possible so you can be approved.
ant r
ant

Comment (Efiamotu-1) [inline app/views/groups/new.html.erb:1]: - [ ] You have one test failing as shown in the image below, please kindly fix it 
tf

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @Alejandroq12, 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by topeogunleye — APPROVED** [review]
### Hi @Alejandroq12 , 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- Addressed all changes requested :heavy_check_mark:
 
## [Optional] suggestions:
* [x]  Please check my inline comments.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Alejandroq12** [comment]
Thank you very much for the time and valuable feedback.

**Pull request merged**